### ISSUE TITLE
Detecting wrong objects

### ISSUE THREAD

**Issue by Moaxen** [nicknochnack/RealTimeObjectDetection#3]
Hi. 
I am trying to detect gestures, at this moment only 2 gestures. But I get a lot of false positive detections.
In train folder 67(including 7 background pictures) pics, in test 6 pics.
Model -  ssd_mobilenet_v2_fpnlite 320x320.
TF2 - version 2.4.1.
Usually I go with 50000 steps and batch size 8 or lower. My GPU can't handle bigger batch size.
I changed background some of my images to different colors, having only my hand on some color. 

I tried to add background images as described here #https://github.com/tensorflow/models/issues/3365#issuecomment-661326737 
But I am not sure if it helped because I still have a lot of false positive detections.
Should I have added backgrounds to the test folder as well?

I also tried lowering learning rate (0.07 -> 0.007) . 
Usually I get loss around  0.058 or 0.150.

I started training with learning rate 0.07 for the first 50000 steps, for the second 50000 steps I lowered rate to 0.007. Simply trying to understand if changing it after 50000 will give different result. If it won't work I wanted to try increasing number of pictures.

What should I do to get rid of the false positive detections?

**Comment by nicknochnack** [comment]
What's in the background images? Are there any detectable examples in these?

**Comment by Moaxen** [comment]
`What's in the background images?`
Some random pics (cats, for example) that have nothing to do with my gestures. 
`Are there any detectable examples in these?`
No. I created for this pics .xml files and deleted from there block "Object". I also changed some code in generate_tfrecord.py like in the link above.

**Comment by nicknochnack** [comment]
Hmmm, I don't think those will add much value as there aren't any labels being generated in the tfrecords file. To improve accuracy I'd suggest the following: 
1. Perform image augmentation e.g. brightness, contrast, scaling and cropping
2. Train for longer until you hit a low loss metric for both train and validation
3. Use more photos for each class and under different conditions

**Comment by Moaxen** [comment]
I added around 30 pics for each class, changed some of them(blur, brightness, contrast and etc.).
I checked detection at 0.079(28300 steps). It was good, but still sometimes detected my head as index finger and with accuracy around 50-60%. I continued training for a lot (209000 steps), stopped at 0.030 loss and results are worse than at 0.079 loss. I have some false positive detections, but, luckily, not as much as before. 

Your suggestions helped a lot. Before I asked you I changed background in my pics for different colors, thought it would help somehow.

Now I will try adding more pics(100) to each class and will be stopping and checking. Did not think that at lower loss it will be performing worse.

Also what I need to do to get id of detected class in python script.
